### DISCUSSION TITLE
Changes to pubspec.lock unrelated to updated dependencies

### DISCUSSION THREAD

**Discussion by slosd** [renovatebot/renovate#19608] [Suggest an Idea]
TLDR: Lock file maintenance seems to run `flutter pub upgrade`, but I'd like it to call `flutter pub upgrade A B`, where _A_ and _B_ are the packages updated in the pull request.

I've been trying out lock file maintenance for dart projects that was implemented in https://github.com/renovatebot/renovate/pull/19116 and the results are unfortunately not what I was expecting: For example, a pull request for updating package _A_ includes a change to pubspec.yaml (e.g. `A: ^1.0.7` to `A: ^2.1.0`) which is fine, but also changes to pubspec.lock with versions of packages updated that are completely unrelated to package _A_. It even includes updates to the required flutter and dart version.

I'd have created a bug report if the docs didn't say
> "Maintaining" a lock file means recreating it so that every dependency version within it is updated to the latest. 

so maybe this is the expected behavior. It's not really suitable to my use case though. The pull request is a means to review a new version (or group of versions), so any unrelated dependency update just complicates the review process.

**Comment by viceice** [comment]
lockfile maintenance from renovate is to update all does inside their range.

you should check if range strategy supports update-lockfile.

**Comment by viceice** [comment]
you're right, please open a feature request to support those scenarios.
